ddmrp_product_replace/models/stock_buffer.py

```diff
@@ -149,6 +149,22 @@ def _search_stock_moves_incoming_domain(self, outside_dlt=False):
             )
         return domain
 
+    def _search_stock_moves_qualified_demand_domain(self):
+        domain = super()._search_stock_moves_qualified_demand_domain()
+        if not self.demand_product_ids:
+            return domain
+        index_replace = False
+        for n, clause in enumerate(domain):
+            if isinstance(clause, tuple) and clause[0] == "product_id":
+                index_replace = n
```

Comment:
```suggestion
                index_replace = n
                break
```

Reply:
AFAIS you are replacing a leaf here.
Normally I do something like this which seems more readable:
```py
domain = [x for x in domain if x[0] != "product_id"] +  [("product_id", "in", self.demand_product_ids.ids)]
```

Reply:
Yes, I agree. Added :+1: